### DISCUSSION TITLE
How to get going on SmartEVSE dev?

### DISCUSSION THREAD

**Discussion by ChuckChance** [serkri/SmartEVSE-3#108] [Q&A] [answered]
Hey,

I have a few feature ideas I'd like to get in my EV charger (biggest project being adding my Victron GX as a mainsmeter through modbus) and I'd like to try to get them in myself but I'm wondering how you guys are getting the testing done.

Is there any emulator for the ESP32 card that can be used to test networking/interface code? Or must all the testing be done on the device itself? Obviously anything related to charging would have to be done on the device with an EV being connected.

Thanks

**Comment by dingo35** [comment, marked as answer]
1. You can probably add your Victron in the "custom" menu when adding the mainsmeter
2. I bought an ESP32 device (they are cheap) for testing, but you have to disable al lot of code (e.g. modbus code), and you cannot really charge, hence in practice it is pretty useless, and we all work with our chargers as testing device.
